core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -473,7 +473,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
 
     // Validate spark.executor.extraJavaOptions
     getOption(executorOptsKey).foreach { javaOpts =>
-      if (javaOpts.contains("-Dspark")) {
+      if (javaOpts.contains("-Dspark") && !javaOpts.contains("-Dspark.hadoop.")) {
```

Comment:
Change to core that we need to discuss @mccheah @foxish @kimoonkim 
We need this to ensure that `spark.hadoop.hadoop.security.authentication`is set to simple in the executors because we have discovered that it is not necessary to run kerberos in the driver+executors as only the delegation token is necessary